### PR TITLE
Assorted bugs

### PR DESCRIPTION
**Pull request by kingainformed**
## Description

Fixes for Trello 63, 72, 75, 77, 78 and 86

## Todo's before PR

- [ ] Branch has been rebased against develop
- [ ] Unit tests passed (`make test`)
- [ ] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [ ] PR description describes the overall goals of PR commits
- [ ] Templates have been checked against the relevant user-story

## Migrations

- [ ] Null fields in models specifies default value
- [ ] You generated and applied migrations (`make migrate`)
- [ ] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [ ] Specified reviewers (even if it's yourself)
- [ ] Specified assignees (those who'll merge)
- [ ] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [ ] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

File: payment_app/views/payment.py [hunk 1]
```diff
@@ -398,6 +398,7 @@ def payment_confirmation(request):
 
     variables = {
         'application_id': application_id_local,
+        'id': application_id_local,
```
Comment (LRydeard) [inline payment_app/views/payment.py]: Is 'application_id' used differently from 'id' in variables or is this a duplicate variable?

File: login_app/views/validate_magic_link.py [hunk 2]
```diff
@@ -2,25 +2,25 @@
...
-            return HttpResponseRedirect(self.get_success_url())
+        if api_response.status_code == 200:
+            self.record = api_response.record[0]
+            if not self.link_has_expired():
+                return HttpResponseRedirect(self.get_success_url())
 
-        else:
-            return HttpResponseRedirect(reverse('Link-Used'))
+        # default
+        return HttpResponseRedirect(reverse('Link-Used'))
```
Comment (Thomas-95IS) [inline login_app/views/validate_magic_link.py]: Is there also a 'Bad-Link' page? Would that be more appropriate here, if it exists?

File: payment_app/views/payment.py (hunk 1 above)
Comment (kingainformed) [reply payment_app/views/payment.py]: `id` needs to be surfaced in order for the Gov UK template to recognise the application - I'm not sure where `application_id` is being used...

File: first_aid_app/views/summary.py [hunk 3]
```diff
@@ -28,5 +28,6 @@ def get_context_data(self, **kwargs):
         application_id = self.request.GET['id']
         context['link_url'] = build_url(self.success_url_name, get={'id': application_id})
         context['application_id'] = self.request.GET['id']
+        context['id'] = self.request.GET['id']
```
Comment (Thomas-95IS) [inline first_aid_app/views/summary.py]: Why do we have both `id` and `application_id` defined?

File: childcare_address_app/views/base.py [hunk 4]
```diff
@@ -55,6 +55,7 @@ def get_context_data(self, **kwargs):
         context = super(BaseTemplateView, self).get_context_data(**kwargs)
         app_id = self.request.GET.get('id')
         context['application_id'] = app_id
+        context['id'] = app_id
```
Comment (Thomas-95IS) [inline childcare_address_app/views/base.py]: Why do we have both `application_id` and `id`?

File: declaration_app/views/master_summary.py [hunk 5]
```diff
@@ -36,32 +39,33 @@ def generate_links(self, json, app_id):
                     table['link'] = reverse(table['reverse']) + '?id=' + app_id
         return json
 
-    def load_json(self, app_id, ordered_models, recurse):
+    def load_json(self, app_id, section_key, section_names, recurse):
         """
         Dynamically builds a JSON to be consumed by the HTML summary page
         :param app_id: the id of the application being handled
-        :param ordered_models: the models to be built for the summary page
+        :param section_names: the models to be built for the summary page
```
Comment (Thomas-95IS) [inline declaration_app/views/master_summary.py]: Missing `section_key` from params list.

File: login_app/views/validate_magic_link.py (hunk 2 above)
Comment (kingainformed) [reply login_app/views/validate_magic_link.py]: I did think of that, but actually the link-used and the bad-link page are more or less identical. The missing page would be the link expired page a conditional level up.

**Pull request merged**